src/OpenTelemetry.Extensions.Hosting/README.md

```diff
@@ -83,6 +83,28 @@ app.Run();
 A fully functional example can be found
 [here](https://github.com/open-telemetry/opentelemetry-dotnet/tree/main/examples/AspNetCore).
 
+### Resources
+To dynamically add resources at startup from the dependency injection you can provide an `IResourceDetector`.
+To make use of it add it to the dependency injection and they you can use the `ISerivceProvider` add it to OpenTelemetry:
+```csharp
+public class MyResourceDetector : IResourceDetector
+{
+    public Resource Detect()
+    {
+        return ResourceBuilder.CreateEmpty()
+            .AddService("your service name")
+            .Build();
+    }
+}
+
+services.AddSingleton<MyResourceDetector>();
+
+services.AddOpenTelemetry()
+    .ConfigureResource(builder =>
+        builder.AddDetector(sp => sp!.GetRequiredService<MyResourceDetector>()))
+    WithTracing(builder => builder.AddConsoleExporter());
```

Comment:
add Metrics too, to make it very clear to users that the resource is applied to traces and metrics both.

Reply:
nit: You should be able to remove the `!` on the `sp` param because it is no longer nullable:

Before: `sp!.GetRequiredService<MyResourceDetector>()`
Updated: `sp.GetRequiredService<MyResourceDetector>()`

Reply:
Done